src/Microsoft.Extensions.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -100,7 +100,16 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                 return;
             }
 
-            propertyValue = BindInstance(property.PropertyType, propertyValue, config.GetSection(property.Name));
+            // if binding IConfigurationSection, break early
+            if (typeof(IConfigurationSection) == property.PropertyType)
```

Comment:
Nit: I think this could be made slightly simpler if we moved the logic into `BindInstance()` E.g. at the start it could do:

``` C#
private static object BindInstance(Type type, object instance, IConfiguration config)
{
    if (type == typeof(IConfigurationSection))
    {
        return config;
    } 
...
```

Then `BindProperty()` would remain untouched.